### PR TITLE
Fix the logic of running the BT discovery thread

### PR DESCRIPTION
**Pull request by romanvm**
This PR fixes the logic of running the Bluetooth discovery thread. It also includes some refactoring of the BT module.

### CODE REVIEW

**Comment by chewitt** [comment]
Functional tested on RPi4 without issues

File: resources/lib/modules/bluetooth.py [hunk 1]
```diff
@@ -677,18 +678,18 @@ def stopped(self, value):
...
         self._stop_event.clear()
         while not self.stopped and not oe.xbmcm.abortRequested():
             current_time = time.time()
-            if current_time > self.last_run + 5:
-                if self.main_menu.getSelectedItem().getProperty('modul') != 'bluetooth' or not hasattr(oe.dictModules['bluetooth'], 'discovery_thread'):
-                    oe.dictModules['bluetooth'].menu_connections()
+            if self.main_menu.getSelectedItem().getProperty('modul') == 'bluetooth':
+                if current_time > self.last_run + 5:
+                    self.parent.discover_devices()
                 self.last_run = current_time
```
Comment (mglae) [inline resources/lib/modules/bluetooth.py:691]: Is this correct? IMO self.parent.discover_devices() will only be executed once at start.
```suggestion
                    self.last_run = current_time
```

**Comment by mglae** [comment]
A minor - maybe not related to this PR.

When entering/exiting the BT tab very seldom an error is triggered like http://ix.io/3r1g

File: resources/lib/modules/bluetooth.py (hunk 1 above)
Comment (romanvm) [reply resources/lib/modules/bluetooth.py:691]: You are correct. I have changed the condition.

**Comment by romanvm** [comment]
@mglae 
> A minor - maybe not related to this PR.
> When entering/exiting the BT tab very seldom an error is triggered like http://ix.io/3r1g

I believe that some race condition happens from time to time but I haven't been able to track it yet. As you can see at line 359 there is a check if a device is in current list items but at line 362 the device is not there already, so something has changed `self.listItems` in between.

**Comment by romanvm** [comment]
Actually I was thinking of running all `discover_devices` method under a thread lock but I'm hesitant because it looks like a big overkill.

**Comment by mglae** [comment]
Agree. Thank you for looking at it.

**Pull request merged**